lib/hawkular/alerts/alerts_api.rb

```diff
@@ -523,8 +523,24 @@ def to_h
   # server and not 'rubyfied'. So 'alertId' and not 'alert_id'
   # Check http://www.hawkular.org/docs/rest/rest-alerts.html#Alert for details
   class Alert < OpenStruct
+    attr_accessor :lifecycle
+
     def initialize(alert_hash)
       super(alert_hash)
+      @lifecycle = alert_hash['lifecycle']
+    end
+
+    def ackBy
+      statusBy('ACKNOWLEDGED')
+    end
+
+    def resolvedBy
+      statusBy('RESOLVED')
+    end
+
+    def statusBy(status)
+      a = @lifecycle.nil? ? [] : @lifecycle.select { |l| l['status'].eql? status }
+      a.empty? ? '' : a.last['user']
```

Comment:
I guess it would be better to return `nil` instead of empty string.
That would help to know when an Alert hasn't been acknowledged.
`alert.resolvedBy.nil?`